tests/phpunit/Civi/API/V4/ContactTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace Civi\API\V4;
+use Civi\Api4\CustomGroup;
+use Civi\Api4\CustomField;
+use Civi\Api4\Contact;
+
+// fixme - what am I doing wrong to need this line?
+require_once 'UnitTestCase.php';
+
+/**
+ * @group headless
+ */
+class EntityTest extends UnitTestCase {
+
+  public function testGetWithCustomData() {
+    $customGroup = CustomGroup::create()
+      ->setCheckPermissions(FALSE)
+      ->setValue('name', 'CustomContactFields')
+      ->setValue('title', 'CustomContactFields')
+      ->setValue('extends', 'Contact')
```

Comment:
@mickadoo made an interesting point that the BAO expects an array for `CustomGroup.extends`.

All my initial reactions on hearing the description were wrong. Understanding this field requires reading at the actual code, e.g.
 * https://github.com/civicrm/civicrm-core/blob/4.7.19/xml/schema/Core/CustomGroup.xml#L46
 * https://github.com/civicrm/civicrm-core/blob/4.7.19/api/v3/CustomGroup.php#L50
 * https://github.com/civicrm/civicrm-core/blob/4.7.19/CRM/Core/BAO/CustomGroup.php#L73

What seems to be going on is that the `civicrm_custom_group.extends` field is legitimately modeled and stored as a single string, but `CustomGroup` BAO expects this signature:

```
$params['extends'] = array(0 => string $baseEntity, 1 => string|array $childTypes)
```

Element `1` (`childTypes`) is optional -- but (more confusingly), you have a choice of passing `childTypes` via `$params['extends'][1]` or `$params['extends_entity_column_value']`.